lib/eksek_result.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require_relative 'eksek_result'
+
+# Describes a result object to be used for evaluating
+# return values from a command
+class EksekResult
+  def initialize cmd, exit_code, success, stdout, stderr
+    @cmd = cmd
+    @exit_code = exit_code
+    @success = success
+    @stdout = stdout
+    @stderr = stderr
+  end
+
+  attr_reader(:exit_code)
+
+  def stdout
+    @stdout ||= nil
```

Comment:
If I'm not mistaken, if `@stdout` always gets a value set in `initialize`, then it's superfluous to use `||=`. Consider using an `attr_reader` instead.